From a11ccc225a38769dcecaaf10f783389eddfe2c7f Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:58 -0400
Subject: [fs] btrfs: free tmp ulist for qgroup rescan

Message-id: <1407179109-13714-352-git-send-email-sandeen@redhat.com>
Patchwork-id: 87246
O-Subject: [RHEL7.1 PATCH 351/422] Btrfs: free tmp ulist for qgroup rescan
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Josef Bacik <jbacik@fb.com>

commit 2a10840945a86dfa2356bce526ae78cd4c1a356e
Author: Josef Bacik <jbacik@fb.com>
Date:   Tue May 20 09:23:31 2014 -0400

    Btrfs: free tmp ulist for qgroup rescan

    Memory leaks are bad mmkay?

    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/qgroup.c b/fs/btrfs/qgroup.c
index 73f706c..cf5aead 100644
--- a/fs/btrfs/qgroup.c
+++ b/fs/btrfs/qgroup.c
@@ -2478,6 +2478,7 @@ static void btrfs_qgroup_rescan_worker(struct btrfs_work *work)
 out:
  kfree(scratch_leaf);
  ulist_free(qgroups);
+ ulist_free(tmp);
  btrfs_free_path(path);
 
  mutex_lock(&fs_info->qgroup_rescan_lock);
-- 
1.7.1